src/ansys/grantami/jobqueue/_models.py

```diff
@@ -351,12 +416,18 @@ def _add_file(self, file_obj: Union[str, pathlib.Path], type_: _FileType) -> Non
             If the file object is not a string or ``pathlib.Path`` object.
         """
         if isinstance(file_obj, pathlib.Path):
-            new_file = _JobFile.from_pathlib_path(file_type=type_, path=file_obj)
+            new_file = _JobFile(
+                path=file_obj,
+                file_type=type_,
+            )
         elif isinstance(file_obj, str):
-            new_file = _JobFile.from_string(file_type=type_, file_path=file_obj)
+            new_file = _JobFile(path=pathlib.Path(file_obj), file_type=type_)
+        elif isinstance(file_obj, JobFile):
+            new_file = _JobFile.from_job_file(file_obj, file_type=type_)
         else:
+            # TODO add JobFile to err msg
             raise TypeError(
-                "file_obj must be a pathlib.Path, BinaryIO, or str object. "
+                "file_obj must be a pathlib.Path, BinaryIO, or str object. "  # TODO binaryIO? what handles it?
```

Comment:
I think we pulled support for binaryIO before releasing, and just never removed it from the error message.